flair/embeddings.py

```diff
@@ -1937,3 +1937,47 @@ def _add_embeddings_internal(self, sentences: List[Sentence]):
                     sentence.set_embedding(embedding.name, sentence[1]._embeddings[embedding.name])
 
         return sentences
+
+
+class NILCEmbeddings(WordEmbeddings):
+
+    def __init__(self, embeddings: str, model: str = 'skip', size: int = 100):
+        """
+        Initializes portuguese classic word embeddings trained by NILC Lab (http://www.nilc.icmc.usp.br/embeddings).
+        Constructor downloads required files if not there.
+        :param embeddings: one of: 'fasttext', 'glove', 'wang2vec' or 'word2vec'
+        :param model: one of: 'skip' or 'cbow'. This is not applicable to glove.
+        :param size: one of: 50, 100, 300, 600 or 1000.
+        """
+
+        base_path = 'http://143.107.183.175:22980/download.php?file=embeddings/'
+
+        cache_dir = Path('embeddings')
+
+        # GLOVE embeddings
+        if embeddings.lower() == 'glove':
+            cached_path(f'{base_path}{embeddings}/{embeddings}_s{size}.zip', cache_dir=cache_dir)
+            embeddings = cached_path(f'{base_path}{embeddings}/{embeddings}_s{size}.zip', cache_dir=cache_dir)
+
+        elif embeddings.lower() in ['fasttext', 'wang2vec', 'word2vec']:
+            cached_path(f'{base_path}{embeddings}/{model}_s{size}.zip', cache_dir=cache_dir)
+            embeddings = cached_path(f'{base_path}{embeddings}/{model}_s{size}.zip', cache_dir=cache_dir)
```

Comment:
I realized that there's a problem here. All the zip files at their repository have the same name pattern: {model}_s{size}.zip. So how can I cache these using different names for different embedding types? For an example: at the cache_dir I would have a word2vec_skip_s100.zip and wang2vec_skip_s100.zip.

Reply:
Maybe you could add a model signature to the `cache_dir` path, e.g. `cache_dir = Path('embeddings') / embeddings.lower() / f'{model}_s{size}'?

Reply:
OK, done. Thanks for the tip!